core/events/events.js

```diff
@@ -218,101 +303,99 @@ Blockly.Events.BumpEvent;
  * appear connected.
  * @const
  */
-Blockly.Events.BUMP_EVENTS = [
-  Blockly.Events.BLOCK_CREATE,
-  Blockly.Events.BLOCK_MOVE,
-  Blockly.Events.COMMENT_CREATE,
-  Blockly.Events.COMMENT_MOVE
-];
+const BUMP_EVENTS = [BLOCK_CREATE, BLOCK_MOVE, COMMENT_CREATE, COMMENT_MOVE];
+exports.BUMP_EVENTS = BUMP_EVENTS;
 
 /**
  * List of events queued for firing.
- * @private
  */
-Blockly.Events.FIRE_QUEUE_ = [];
+const FIRE_QUEUE = [];
+/** @private */
+exports.FIRE_QUEUE_ = FIRE_QUEUE;
 
 /**
  * Create a custom event and fire it.
- * @param {!Blockly.Events.Abstract} event Custom data for event.
+ * @param {!Abstract} event Custom data for event.
  */
-Blockly.Events.fire = function(event) {
-  if (!Blockly.Events.isEnabled()) {
+const fire = function(event) {
+  if (!isEnabled()) {
     return;
   }
-  if (!Blockly.Events.FIRE_QUEUE_.length) {
+  if (!FIRE_QUEUE.length) {
     // First event added; schedule a firing of the event queue.
-    setTimeout(Blockly.Events.fireNow_, 0);
+    setTimeout(fireNow, 0);
   }
-  Blockly.Events.FIRE_QUEUE_.push(event);
+  FIRE_QUEUE.push(event);
 };
+exports.fire = fire;
 
 /**
  * Fire all queued events.
- * @private
  */
-Blockly.Events.fireNow_ = function() {
-  var queue = Blockly.Events.filter(Blockly.Events.FIRE_QUEUE_, true);
-  Blockly.Events.FIRE_QUEUE_.length = 0;
-  for (var i = 0, event; (event = queue[i]); i++) {
+const fireNow = function() {
+  const queue = filter(FIRE_QUEUE, true);
+  FIRE_QUEUE.length = 0;
+  for (let i = 0, event; (event = queue[i]); i++) {
     if (!event.workspaceId) {
       continue;
     }
-    var workspace = Blockly.Workspace.getById(event.workspaceId);
-    if (workspace) {
-      workspace.fireChangeListener(event);
+    const Workspace = goog.module.get('Blockly.Workspace');
+    const eventWorkspace = Workspace.getById(event.workspaceId);
+    if (eventWorkspace) {
+      eventWorkspace.fireChangeListener(event);
     }
   }
 };
+/** @private */
+exports.fireNow_ = fireNow;
 
 /**
  * Filter the queued events and merge duplicates.
- * @param {!Array<!Blockly.Events.Abstract>} queueIn Array of events.
+ * @param {!Array<!Abstract>} queueIn Array of events.
  * @param {boolean} forward True if forward (redo), false if backward (undo).
- * @return {!Array<!Blockly.Events.Abstract>} Array of filtered events.
+ * @return {!Array<!Abstract>} Array of filtered events.
  */
-Blockly.Events.filter = function(queueIn, forward) {
-  var queue = queueIn.slice();  // Shallow copy of queue.
+const filter = function(queueIn, forward) {
+  let queue = queueIn.slice();  // Shallow copy of queue.
```

Comment:
I think `queue` is `const`.  The reverse is an in place operation.

Reply:
It gets reassigned on line 409.